frontend/lib/resolution/scope-queries.cpp

```diff
@@ -1652,48 +1664,67 @@ getSymbolsAvailableInScopeQuery(Context* context,
 
   std::map<UniqueString, BorrowedIdsWithName> toReturn;
 
-  auto allowedByVisibility = [inVisibilitySymbols](UniqueString name, UniqueString& renameTo) {
+  auto allowedByVisibility = [inVisibilitySymbols](UniqueString name,
+                                                   UniqueString& renameTo,
+                                                   bool isSymbolItself) {
     renameTo = name;
 
-    if (!inVisibilitySymbols) return true;
+    if (!inVisibilitySymbols) return !isSymbolItself;
     auto kind = inVisibilitySymbols->kind();
 
     if (kind == VisibilitySymbols::ALL_CONTENTS) {
-      return true;
+      // ALL_CONTENTS brings in the contents, but not the symbol itself.
+      return !isSymbolItself;
     }
 
-    if (kind == VisibilitySymbols::ONLY_CONTENTS ||
-        kind == VisibilitySymbols::CONTENTS_EXCEPT) {
-      auto& namePairs = inVisibilitySymbols->names();
+    // Should we even bother checking the names?
```

Comment:
Maybe remove this comment? Checking the names seems reasonable to handle `use X as Y` etc. And it does check the names.